frontend/sw360-portlet/src/main/java/com/siemens/sw360/portal/tags/CompareTodos.java

```
@@ -0,0 +1,194 @@
+import com.siemens.sw360.datahandler.thrift.licenses.Obligation;
+import com.siemens.sw360.datahandler.thrift.licenses.Todo;
+import org.apache.thrift.meta_data.FieldMetaData;
+
+import javax.servlet.jsp.JspException;
+import javax.servlet.jsp.JspWriter;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static com.google.common.collect.ImmutableList.copyOf;
+import static com.siemens.sw360.datahandler.common.CommonUtils.nullToEmptyList;
+import static com.siemens.sw360.portal.tags.TagUtils.*;
+
+/**
+ * Display the fields that have changed from old to update
+ *
+ * @author birgit.heydenreich@tngtech.com
+ */
+public class CompareTodos extends NameSpaceAwareTag {
+    public static final List<Todo._Fields> RELEVANT_FIELDS = FluentIterable
+            .from(copyOf(Todo._Fields.values())).filter(field -> isFieldRelevant(field)).toList();
+
+    private List<Todo> old;
+    private List<Todo> update;
+    private String tableClasses = "";
+    private String idPrefix = "";
+
+    public void setOld(List<Todo> old) {
+        this.old = nullToEmptyList(old);
+    }
+
+    public void setUpdate(List<Todo> update) {
+        this.update = nullToEmptyList(update);
+    }
+
+    public void setTableClasses(String tableClasses) {
+        this.tableClasses = tableClasses;
+    }
+
+    public void setIdPrefix(String idPrefix) {
+        this.idPrefix = idPrefix;
+    }
+
+    public int doStartTag() throws JspException {
+
+        JspWriter jspWriter = pageContext.getOut();
+        StringBuilder display = new StringBuilder();
+        String namespace = getNamespace();
+
+        try {
+            renderTodos(display, old, update);
+
+            String renderString = display.toString();
+
+            if (Strings.isNullOrEmpty(renderString)) {
+                renderString = "<h4> No changes in TODOs </h4>";
+            }
+
+            jspWriter.print(renderString);
+        } catch (Exception e) {
```

Why do you catch all exceptions here?